src/main/java/io/cdap/plugin/salesforce/authenticator/Authenticator.java

```diff
@@ -89,12 +109,41 @@ public static OAuthInfo getOAuthInfo(AuthenticatorCredentials credentials) throw
       if (!Strings.isNullOrEmpty(authResponse.getError())) {
         throw new IllegalArgumentException(
           String.format("Cannot authenticate to Salesforce with given credentials. ServerResponse='%s'",
-                  response));
+                        response));
       }
 
       return new OAuthInfo(authResponse.getAccessToken(), authResponse.getInstanceUrl());
     } finally {
       httpClient.stop();
     }
   }
+
+  /**
+   * Set a proxy to http client based on user input.
+   *
+   * @param credentials Credentials contain the proxy server details set in config.
+   * @param httpClient  httpClient to be used to call salesforce APIs
+   */
+  public static void setProxy(AuthenticatorCredentials credentials, HttpClient httpClient) {
+    try {
+      if (!Strings.isNullOrEmpty(credentials.getProxyUrl())) {
+        URI proxyUrl = new URI(credentials.getProxyUrl());
+        ProxyConfiguration proxyConfig = httpClient.getProxyConfiguration();
+        HttpProxy proxy = new HttpProxy(proxyUrl.getHost(), proxyUrl.getPort());
+        proxyConfig.getProxies().add(proxy);
+        // set auth store to http client if proxy is secured with username and password.
+        if (!Strings.isNullOrEmpty(credentials.getProxyUsername()) &&
+          !Strings.isNullOrEmpty(credentials.getProxyPassword())) {
+          AuthenticationStore store = new HttpAuthenticationStore();
+          store.addAuthentication(new BasicAuthentication(proxyUrl, Authentication.ANY_REALM,
```

Comment:
Is the authentication part tested with Squid proxy? What does `ANY_REALM` signify here?